ovos_chromadb_embeddings/__init__.py

```
@@ -2,112 +1,409 @@
+    def _get_collection_instance(self, collection_name: Optional[str]) -> chromadb.api.models.Collection.Collection:
+        """
+        Helper method to get the ChromaDB collection instance.
+        If collection_name is None, returns the default collection.
+        Raises ValueError if the collection does not exist.
+        """
+        name = collection_name or self.default_collection_name
+        try:
+            return self.client.get_collection(name=name)
+        except Exception as e:
+            # ChromaDB raises an exception if collection not found
+            raise ValueError(f"Collection '{name}' not found or could not be accessed: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)